bin/snapshot-backups/snapshot-purge

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+
+# we might be running in circleci
+if [ -f /home/circleci/project/env.local ]; then
+  . /home/circleci/project/env.local
+fi
+# we might be running from a local dev machine
+SCRIPT_DIR="$(dirname "$0")"
+if [ -f $SCRIPT_DIR/env.local ]; then
+  . $SCRIPT_DIR/env.local
+fi
+if [ -f ./env.local ]; then
+  . ./env.local
+fi
+
+if [ -f $SCRIPT_DIR/../deploy/includes ]; then
+  . $SCRIPT_DIR/../deploy/includes
+else
+   echo Cannot find $SCRIPT_DIR/../deploy/includes
+   exit 1
+fi
+
+export BUCKET_SOURCE_FOLDER=public
+if [ -f ./bin/snapshot-backups/snapshot-include ]; then
+  . ./bin/snapshot-backups/snapshot-include
+else
+   echo Cannot find ${PWD}/bin/snapshot-backups/snapshot-include
+   exit 1
+fi
+if [ -f $SCRIPT_DIR/../deploy/includes ]; then
+  . $SCRIPT_DIR/../deploy/includes
+else
+   echo Cannot find $SCRIPT_DIR/../deploy/includes
+   exit 1
+fi
+
+# just testing?
+if [ x$1 = x"--dryrun" ]; then
+  echo=echo
+  dryrun=$1
+  shift
+fi
+
+SPACE=${1:-please-provide-space-name-as-first-argument}
+SPACE=$(echo "$SPACE" | tr '[:upper:]' '[:lower:]')
+assertCurSpace $SPACE
+shift
+
+removeTag=""
+removeTicket=""
+download=""
+
+while [ 0 -eq 0 ]; do
+
+  snaptype=$1
+
+  if [ x$snaptype = x ]; then
+    break
+  fi
+  shift
+
+  case "$snaptype" in
+
+      "--tag")
+        removeTag=$1
+        if [ x"$removeTag" == x ]; then
+          echo "Usage:"
+          echo "$0 [--dryrun] <env> [--ticket <eg. USAGOV-9999> | --tag <eg. USAGOV-9999.dev.1111.post-deploy.2020010101] [--download]"
+          echo ""
+          echo "--dryrun MUST be the first argument, or it WILL NOT BE ACKNOWLEGED"
+          echo "--download will download a copy of the snapshot(s) before deleting them"
+          exit 1
+        fi
+      ;;
+
+      "--ticket")
+        removeTicket=$1
+        if [ x"$removeTicket" == x ]; then
+          echo "Usage:"
+          echo "$0 [--dryrun] <env> [--ticket <eg. USAGOV-9999> | --tag <eg. USAGOV-9999.dev.1111.post-deploy.2020010101] [--download]"
+          echo ""
+          echo "--dryrun MUST be the first argument, or it WILL NOT BE ACKNOWLEGED"
+          echo "--download will download a copy of the snapshot(s) before deleting them"
+          exit 1
+        fi
+      ;;
+
+      "--download")
+        download="--download"
+      ;;
+  esac
+
+done
+
+echo $(basename $0)
+echo "Dry run:       $dryrun"
+echo "Download:      $download"
+echo "Remove tag:    $removeTag"
+echo "Remove ticket: $removeTicket"
+
+source bin/cloudgov/get-s3-access storage >/dev/null 2>&1
+
+export site_available_tags=$(aws s3 ls s3://$S3_BUCKET/web-backup/ | awk '{print $NF}' | sed 's/[\/ ]*$//g')
+export public_available_tags=$(aws s3 ls s3://$S3_BUCKET/$BUCKET_BACKUP_FOLDER/ | awk '{print $NF}' | sed 's/[\/ ]*$//g')
+export db_available_tags=$(aws s3 ls s3://$S3_BUCKET/db-backup/ | awk '{print $NF}' | sed 's/[\/ ]*$//g' | sed 's/\.sql\.gz//g')
+
+if [ x$removeTicket != x ]; then
```

Comment:
Here's my non-documentation issue: Unless I'm missing something, the --ticket and --tag arguments do exactly the same thing. I had expected them --ticket to be a match on just the ticket, and --tag to be a full match. (And then I thought, why don't we just call them that? So if you want to prune all the pre-deploy snapshots, you go for it and use "partial match.")

So my request is to either remove the tag/ticket distinction and just call it something like "substring", or to actually make a substring vs fullmatch distinction.

Reply:
OK --full <string> and --partial <string>.
Argument handling improved a bit, in terms of being verbose in dubious situations.